doc/how_to/custom_components/esm/custom_layout.md

```diff
@@ -1,108 +1,92 @@
-            self._object,
-            **layoutable_params,
-        )
+.gutter {
+    background-color: #eee;
+    background-repeat: no-repeat;
+    background-position: 50%;
+}
 
-    def __panel__(self):
-        return self._layout
+.gutter.gutter-horizontal {
+    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAeCAYAAADkftS9AAAAIklEQVQoU2M4c+bMfxAGAgYYmwGrIIiDjrELjpo5aiZeMwF+yNnOs5KSvgAAAABJRU5ErkJggg==');
+    cursor: col-resize;
+}
+"""
 
-    @pn.depends("object")
-    def _object(self):
-        return self.object
 
+class SplitJS(JSComponent):
 
-dial = pn.widgets.Dial(
-    name="°C",
-    value=37,
-    format="{value}",
-    colors=[(0.40, "green"), (1, "red")],
-    bounds=(0, 100),
-)
-py_layout = SingleObjectLayout(
-    object=dial,
-    name="Temperature",
-    styles={"border": "2px solid lightgray"},
-    sizing_mode="stretch_width",
-)
-py_layout.servable()
-```
+    left = Child()
+    right = Child()
 
-:::
+    _esm = """
+    import Split from 'https://esm.sh/split.js@1.6.5'
 
-:::{tab-item} `JSComponent`
+    const splitDiv = document.createElement('div');
+    splitDiv.className = 'split';
 
-```{pyodide}
-import panel as pn
-from panel.custom import JSComponent, Child
+    const split0 = document.createElement('div');
+    splitDiv.appendChild(split0);
 
-pn.extension()
+    let split1 = document.createElement('div');
+    splitDiv.appendChild(split1);
 
-class SingleObjectLayout(JSComponent):
-    object = Child(allow_refs=False)
+    Split([split0, split1])
 
-    _esm = """
-export function render({ model }) {
-    const containerID = `id-${crypto.randomUUID()}`;;
-    const div = document.createElement("div");
-    div.innerHTML = `
-    <div>
-        <h1>Temperature</h1>
-        <h2>A measurement from the sensor</h2>
-        <div id="${containerID}">...</div>
-    </div>`;
-    const container = div.querySelector(`#${containerID}`);
-    container.appendChild(model.get_child("object"))
-    return div;
-}
-"""
+    export function render({ model }) {
+        split0.append(model.get_child("left"))
+        split1.append(model.get_child("right"))
+      return splitDiv
+    }"""
 
-dial = pn.widgets.Dial(
-    name="°C",
-    value=37,
-    format="{value}",
-    colors=[(0.40, "green"), (1, "red")],
-    bounds=(0, 100),
-)
-js_layout = SingleObjectLayout(
-    object=dial,
-    name="Temperature",
-    styles={"border": "2px solid lightgray"},
+    _stylesheets = [CSS]
+
+
+pn.extension("codeeditor")
+
+split_js = SplitJS(
+    left=pn.widgets.CodeEditor(
+        value="Left!",
+        sizing_mode="stretch_both",
+        margin=0,
+        theme="monokai",
+        language="python",
+    ),
+    right=pn.widgets.CodeEditor(
+        value="right",
+        sizing_mode="stretch_both",
+        margin=0,
+        theme="monokai",
+        language="python",
+    ),
+    height=500,
     sizing_mode="stretch_width",
 )
-js_layout.servable()
+split_js.servable()
```

Comment:
When running the example i see the below in the terminal.

```bash
/home/jovyan/repos/private/panel/panel/viewable.py:302: ParamFutureWarning: Parameter 'left' on <class 'bokeh_app_4ceaa2f3922c4160bc04a620cd2dd69f.SplitJS'> is being given a valid parameter reference CodeEditor(language='python', margin=0, sizing_mode='stretch_both', theme='monokai', value='Left!') but is implicitly allow_refs=False. In future allow_refs will be enabled by default and the reference CodeEditor(language='python', margin=0, sizing_mode='stretch_both', theme='monokai', value='Left!') will be resolved to its underlying value Left!. Please explicitly set allow_ref on the Parameter definition to declare whether references should be resolved or not.
```

Reply:
Fixed.